protobufs/livekit_agent.proto

```diff
@@ -21,154 +21,174 @@ option ruby_package = "LiveKit::Proto";
-    WS_FULL = 1;
+  WS_AVAILABLE = 0;
+  WS_FULL = 1;
 }
 
 enum JobStatus {
-    JS_UNKNOWN = 0;
-    JS_SUCCESS = 1;
-    JS_FAILED = 2;
+  JS_PENDING = 0;
+  JS_STARTED = 1;
+  JS_SUCCESS = 2;
+  JS_FAILED = 3;
 }
 
 message SimulateJobRequest {
-    JobType type = 1;
-    Room room = 2;
-    ParticipantInfo participant = 3;
+  JobType type = 1;
+  Room room = 2;
+  ParticipantInfo participant = 3;
 }
 
 message WorkerPing {
-    int64 timestamp = 1;
+  int64 timestamp = 1;
 }
 
 message WorkerPong {
-    int64 last_timestamp = 1;
-    int64 timestamp = 2;
+  int64 last_timestamp = 1;
+  int64 timestamp = 2;
 }
 
 message RegisterWorkerRequest {
-    JobType type = 1;
-    //string worker_id = 2;
-    string version = 3;
-    string name = 4;
-    uint32 ping_interval = 5;
-    optional string namespace = 6;
-    ParticipantPermission allowed_permissions = 7;
+  JobType type = 1;
+  //string worker_id = 2;
+  string version = 3;
+  string name = 4;
+  uint32 ping_interval = 5;
+  optional string namespace = 6;
+  ParticipantPermission allowed_permissions = 7;
 }
 
 message RegisterWorkerResponse {
-    string worker_id = 1;
-    ServerInfo server_info = 3;
+  string worker_id = 1;
+  ServerInfo server_info = 3;
 }
 
 message MigrateJobRequest {
-    string job_id = 1;
+  string job_id = 1;
 }
 
 message AvailabilityRequest {
-    Job job = 1;
+  Job job = 1;
 
-    // True when the job was previously assigned to another worker but has been 
-    // migrated due to different reasons (e.g. worker failure, job migration)
-    bool resuming = 2;
+  // True when the job was previously assigned to another worker but has been 
+  // migrated due to different reasons (e.g. worker failure, job migration)
+  bool resuming = 2;
 }
 
 
 message AvailabilityResponse {
-    string job_id = 1;
-    bool available = 2;
-    bool supports_resume = 3;
+  string job_id = 1;
+  bool available = 2;
+  bool supports_resume = 3;
 
-    string participant_name = 4;
-    string participant_identity = 5;
-    string participant_metadata = 6;
+  string participant_name = 4;
+  string participant_identity = 5;
+  string participant_metadata = 6;
 }
 
 message UpdateJobStatus {
-    string job_id = 1;
+  string job_id = 1;
 
-    // The worker can indicate the job end by either specifying SUCCESS or FAILED
-    optional JobStatus status = 2;
+  // The worker can indicate the job end by either specifying SUCCESS or FAILED
+  optional JobStatus status = 2;
 
-    // metadata shown on the dashboard, useful for debugging
-    string error = 3;
+  // metadata shown on the dashboard, useful for debugging
+  string error = 3;
 
-    // the metadata can be updated multiple times
-    optional string metadata = 4;
+  // the metadata can be updated multiple times
+  // optional string metadata = 4 [deprecated=true];
     
-    // The load that the job currently has on the worker
-    float load = 5;
+  // The load that the job currently has on the worker
+  float load = 5;
```

Comment:
as convenient as it would be to have this does the agent client have any way to give us job load info? isn't most of the work done in a single process? @theomonnom ?

Reply:
Egress runs each worker in a separate process and reports load for each ingress job separately, but that took a not insignificant amount of effort to implement and may be a too high burden to put on all agent implementations. I agree we should delete it unless @theomonnom has some further input on how this would be used?